### ISSUE TITLE
SEEK with UNION doesn't work correctly

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#7459]
The following query doesn't work correctly. There are book IDs in `(1, 2, 3, 4)`, but the following test fails:

```java
assertEquals(asList(4, 11),
    create().select(TBook_ID())
            .from(TBook())
            .union(
                select(TBook_ID().add(10))
               .from(TBook()))
            .orderBy(TBook_ID())
            .seek(3)
            .limit(2)
            .fetch(TBook_ID()));
```

The generated query is:

```sql
(
  select "PUBLIC"."T_BOOK"."ID"
  from "PUBLIC"."T_BOOK"
  where (
    1 = 1
    and "PUBLIC"."T_BOOK"."ID" > 3
  )
)
union (
  select ("PUBLIC"."T_BOOK"."ID" + 10)
  from "PUBLIC"."T_BOOK"
)
order by "ID"
limit 2
```

But it should be:

```sql
select * 
from (
  (
    select "PUBLIC"."T_BOOK"."ID"
    from "PUBLIC"."T_BOOK"
  )
  union (
    select ("PUBLIC"."T_BOOK"."ID" + 10)
    from "PUBLIC"."T_BOOK"
  )
) x
where "ID" > 3
order by "ID"
limit 2
```

**SEEK BEFORE**

`SEEK BEFORE` is a bit more complicated (see #6380), as it needs to nested orderings. The following query

```java
assertEquals(asList(4, 11),
    create().select(TBook_ID())
            .from(TBook())
            .union(
                select(TBook_ID().add(10))
               .from(TBook()))
            .orderBy(TBook_ID())
            .seek(12)
            .limit(2)
            .fetch(TBook_ID()));
```

Will need to be emulated as:

```sql
select *
from (
  select * 
  from (
    (
      select "PUBLIC"."T_BOOK"."ID"
      from "PUBLIC"."T_BOOK"
    )
    union (
      select ("PUBLIC"."T_BOOK"."ID" + 10)
      from "PUBLIC"."T_BOOK"
    )
  ) x
  where "ID" < 11
  order by "ID" desc -- Inverse ordering in derived table
  limit 2
) x
order by "ID"
```

**Comment by lukaseder** [comment]
This fix and related `SEEK` fixes are too complex to merge to jOOQ 3.10. Will fix for jOOQ 3.11 only.

**Comment by lukaseder** [comment]
The `SEEK BEFORE` clause is no different for unions than for non-union queries, so it will be fixed with #6380
